### DISCUSSION TITLE
[Issue] Update notes for AZD Core -> AVM (Keyvault)

### DISCUSSION THREAD

**Discussion by tonybaloney** [Azure/azure-dev#4106] [Q&A]
Opening this issue as a note on how to upgrade from key vault in core/ to AVM.

1. AVM uses Premium SKU by default, AZD Core used standard. You must override this setting
2. AVM will enable purge protection and soft delete by default (AZD core did not)
3. RBAC authorisation is enabled by default in AVM
4. The key vault URI output name is `uri` instead of `endpoint`
5. AVM uses types for settings like secrets
6. Keyvault secrets are a parameter for the key vault resource
7. Keyvault secrets do not have the same defaults as AZD core, specifically the `exp` and `nbf` parameters for secrets were defaulting to `0` in AZD core and in AVM default to whatever the Azure API default is
8. AVM has a parameter for role assignments when using RBAC, but this will probably create a circular dependency with an application that needs the key vault URI.

**Comment by tonybaloney** [comment]
Example upgrade https://github.com/Azure-Samples/Azure-Python-Standardization-Template-Generator/pull/84

**Comment by jongio** [comment]
Tony we are in the process of moving from infra/core to AVM:

Here's the issue: https://github.com/Azure/azure-dev/issues/3327

Here's the PR: https://github.com/Azure/azure-dev/pull/3976
